ml-agents/mlagents/trainers/subprocess_env_manager.py

```diff
@@ -175,7 +180,20 @@ def external_brains():
                 _send_response(EnvironmentCommand.EXTERNAL_BRAINS, external_brains())
             elif req.cmd == EnvironmentCommand.RESET:
                 for k, v in req.payload.items():
-                    env_parameters.set_float_parameter(k, v)
+                    if isinstance(v, float):
+                        env_parameters.set_float_parameter(k, v)
+                    elif isinstance(v, UniformSettings):
```

Comment:
I really feel like these methods should live outside of the `subprocess_env_manager`. Especially since now it doesn't work with the `simple_env_manager`. Maybe in the env_parameters side_channel we can make a `set_sampler_parameters(parameters: ParameterRandomizationSettings)` method?

Reply:
Addressed. good call out.